flag-app/flag-app-project/src/hooks/useApiData.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+
+import apiClient from "../services/api-client";
+import { CanceledError } from "axios";
```

Comment:
I would suggest sorting the imports slightly better:

1. Libraries go on top
2. Your interfaces and components go second
3. CSS imports

Each group would be separated by an empty line, making this import set look better:

´´´
import { useEffect, useState } from "react";
import { CanceledError } from "axios";

import apiClient from "../services/api-client";
import { Country } from "../interfaces";
´´´

Also try to export apiClient in the services/index.ts